vrx_gz/src/ScanDockScoringPlugin.cc

```diff
@@ -0,0 +1,311 @@
+  /// submisison.
+  public: ColorSequenceChecker(const std::vector<std::string> &_expectedColors,
+                               const std::string &_colorSequenceTopic);
+  /// \brief Enable submissions.
+  public: void Enable();
+
+  /// \brief Disable submissions.
+  public: void Disable();
+
+  /// \brief Whether a team submitted the color sequence or not.
+  /// \return True when the submission was received or false otherwise.
+  public: bool SubmissionReceived() const;
+
+  /// \brief Whether a team submitted the color sequence and is correct or not.
+  /// \return True when the team submitted the color sequence and it is correct
+  /// or false otherwise.
+  public: bool Correct() const;
+
+  /// \brief Callback executed when a new color submission is received.
+  /// \param[in] _sequence Contains the submission. The size of the vector
+  /// should be three.
+  private: void OnColorSequence(const msgs::StringMsg_V &_sequence);
+
+  /// \brief The expected color sequence.
+  private: std::vector<std::string> expectedSequence;
+
+  /// \brief Topic where the color sequence should be sent.
+  private: std::string colorSequenceTopic;
+
+  /// \brief The transport node.
+  private: transport::Node node;
+
+  /// \brief Whether the color sequence has been received or not.
+  private: bool colorSequenceReceived = false;
+
+  /// \brief Whether the color sequence received is correct or not.
+  private: bool correctSequence = false;
+};
+
+//////////////////////////////////////////////////
+ColorSequenceChecker::ColorSequenceChecker(
+  const std::vector<std::string> &_expectedColors,
+  const std::string &_colorSequenceTopic)
+  : expectedSequence(_expectedColors),
+    colorSequenceTopic(_colorSequenceTopic)
+{
+}
+
+//////////////////////////////////////////////////
+void ColorSequenceChecker::Enable()
+{
+  this->node.Subscribe(this->colorSequenceTopic,
+    &ColorSequenceChecker::OnColorSequence, this);
+}
+
+//////////////////////////////////////////////////
+void ColorSequenceChecker::Disable()
+{
+  this->node.Unsubscribe(this->colorSequenceTopic);
+}
+
+/////////////////////////////////////////////////
+bool ColorSequenceChecker::SubmissionReceived() const
+{
+  return this->colorSequenceReceived;
+}
+
+/////////////////////////////////////////////////
+bool ColorSequenceChecker::Correct() const
+{
+  return this->correctSequence;
+}
+
+/////////////////////////////////////////////////
+void ColorSequenceChecker::OnColorSequence(const msgs::StringMsg_V &_sequence)
+{
+  gzmsg << "ColorSequenceChecker: Color sequence submission received:"
+        << _sequence.DebugString() << std::endl;
+  std::cout << std::flush;
+
+  // Sanity check: Only one submission is allowed.
+  if (this->colorSequenceReceived)
+  {
+    gzerr << "The color sequence has already been submitted" << std::endl;
+    return;
+  }
+
+  // Sanity check: We should receive three colors.
+  if (_sequence.data_size() < 3u)
```

Comment:
Currently, if we submit a sequence with more than 3 colors, it will just score the first three. I'm wondering if we should change `< 3u` to `!= 3u` to also discard sequences that are too long.

Reply:
Sounds good to me. Changed in [558ef4d](https://github.com/osrf/vrx/pull/537/commits/558ef4d29f251d7756f965fd462c5404cc86305c).